src/Microsoft.Restier.EntityFramework/Submit/ChangeSetPreparer.cs

```diff
@@ -230,6 +230,12 @@ private static object ConvertToEfValue(Type type, object value)
                 return (TimeSpan)timeOfDayValue;
             }
 
+            // In case key is long type, when put an entity, key value will be from key parsing which is type of int
+            if (value is int && type == typeof(long))
```

Comment:
No, as this is only used for key which is parsed by RESTier, and for other update properties, they are parsed by WebAPi, and type is always correct which I have verified.